manifests/params.pp

```diff
@@ -157,6 +157,13 @@
   $puppet_group       = 'puppet'
   $puppetdir          = $puppet::params::dir
 
+  # The puppet-agent package, (puppet 4 AIO) doesn't create a puppet group
+  if versioncmp($::puppetversion, '4.0') > 0 {
```

Comment:
I guess it's OK if it's ensured that the package is installed before the user/group information is potentially changed.